app/src/main/java/com/example/cooky/ui/favorite/FavoriteFragment.kt

```diff
@@ -28,28 +29,38 @@ class FavoriteFragment private constructor() :
         )
     }
 
+    private val likeAdapter = RecipeAdapterVertical {
+        parentFragment?.findNavController()?.navigate(
+            HomeFragmentDirections.actionGlobalRecipeDetail(it.recipeId)
+        )
+    }
+
     private val sharedPreferences: SharedPreferences by inject()
     override val viewModel: FavoriteViewModel by viewModel()
 
     override fun initView() {
         initRecyclerView()
         getRecentlyRecipeIds()
         observeDataViewModel()
+        nestedScrollView.setOnScrollChangeListener(viewModel.onScrollListener)
     }
 
     private fun initRecyclerView() {
         recyclerRecent.adapter = recentAdapter
+        recyclerViewLike.adapter = likeAdapter
     }
 
     private fun getRecentlyRecipeIds() {
         val recentlyIdsString = sharedPreferences.getString(RECENTLY_ID, STRING_NULL)
-        val newRecentlyIds = stringToListInteger(recentlyIdsString!!)
-        if (recentlyIds.isEmpty()) {
-            recentlyIds.addAll(newRecentlyIds)
-            viewModel.getRecentlyRecipes(recentlyIds)
-        } else if (recentlyIds[0] != newRecentlyIds[0]) {
-            recentlyIds = newRecentlyIds.toMutableList()
-            viewModel.getRecentlyRecipes(recentlyIds)
+        if (recentlyIdsString != STRING_NULL) {
+            val newRecentlyIds = stringToListInteger(recentlyIdsString!!)
+            if (recentlyIds.isEmpty()) {
+                recentlyIds.addAll(newRecentlyIds)
+                viewModel.getRecentlyRecipes(recentlyIds)
+            } else if (recentlyIds[0] != newRecentlyIds[0]) {
```

Comment:
Lấy phần tử đầu có thể dùng first thì code dễ đọc hơn.
```suggestion
            } else if (recentlyIds.first != newRecentlyIds.first) {
```

Reply:
oke không biết có th first luôn :))